rest/rest/src/main/java/io/syndesis/rest/v1/handler/connection/ConnectorTemplateHandler.java

```diff
@@ -15,30 +15,87 @@
  */
 package io.syndesis.rest.v1.handler.connection;
 
+import javax.persistence.EntityNotFoundException;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
 import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.UriInfo;
 
 import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import io.syndesis.connector.generator.ConnectorGenerator;
 import io.syndesis.dao.manager.DataManager;
 import io.syndesis.model.Kind;
+import io.syndesis.model.ListResult;
+import io.syndesis.model.connection.Connector;
 import io.syndesis.model.connection.ConnectorTemplate;
+import io.syndesis.model.connection.ConnectorSettings;
 import io.syndesis.rest.v1.handler.BaseHandler;
 import io.syndesis.rest.v1.operations.Getter;
 import io.syndesis.rest.v1.operations.Lister;
 
+import org.springframework.context.ApplicationContext;
 import org.springframework.stereotype.Component;
 
-@Path("/connector-templates")
-@Api(value = "connector-templates")
+@Path("/custom/connectors")
+@Api(tags = {"custom-connectors", "connector-template"})
 @Component
-public class ConnectorTemplateHandler extends BaseHandler implements Lister<ConnectorTemplate>, Getter<ConnectorTemplate> {
+public final class ConnectorTemplateHandler extends BaseHandler
+    implements Lister<ConnectorTemplate>, Getter<ConnectorTemplate> {
 
-    public ConnectorTemplateHandler(final DataManager dataMgr) {
+    private final ApplicationContext applicationContext;
+
+    protected ConnectorTemplateHandler(final DataManager dataMgr, final ApplicationContext applicationContext) {
         super(dataMgr);
+        this.applicationContext = applicationContext;
+    }
+
+    @POST
+    @Path("/{id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    @ApiOperation("Creates a new Connector based on the ConnectorTemplate identified by the provided `id`  and the data given in`customConnector`")
+    @ApiParam(name = "id", example = "swagger-connector-template")
+    @ApiResponses(@ApiResponse(code = 200, response = Connector.class, message = "Newly created Connector"))
+    public Connector create(@PathParam("id") @NotNull final String templateId, final ConnectorSettings connectorSettings) {
+        final ConnectorTemplate connectorTemplate = getDataManager().fetch(ConnectorTemplate.class, templateId);
+
+        if (connectorTemplate == null) {
+            throw new EntityNotFoundException("Connector template: " + templateId);
+        }
+
+        final ConnectorGenerator connectorGenerator = applicationContext.getBean(templateId, ConnectorGenerator.class);
```

Comment:
in https://github.com/syndesisio/syndesis/pull/450/files#diff-89486c5a63a324575071f5699d43aae1R70 you check for `null`, but not here. Not sure what is correct, maybe this duplicate code could be extracted to a single method ? (however not worth the effort if an extra class would need to be introduced only for this)